packages/api-derive/test/e2e/promise.spec.ts

```diff
@@ -100,4 +103,35 @@ describe.skip('derive e2e', () => {
       done();
     });
   });
+
+  describe.only('verifies derive.staking.unlocking',() => {
+    const BOND_VALUE = 10;
+    const UNBOND_VALUE = 1;
+    const ALICE_STASH = '5GNJqTPyNqANBkUVMN1LPPrxXnFouWXoe2wNSmmEoLctxiZY';
+    const ALICE = '5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY';
+    const keyring = testKeyring();
+    const aliceStashPair = keyring.getPair(ALICE_STASH);
+    const alicePair = keyring.getPair(ALICE);
+
+    it('bondsExtra funds for Alice Stash', () => {
+      return api.tx.staking.bondExtra(BOND_VALUE)
+        .signAndSend(aliceStashPair, ({ events = [], status }) => {
+          expect(status.type).toMatch(/Ready/i);
+        });
+    });
+
+    it('unbonds dots for Alice (from Alice Stash)', () => {
+      return api.tx.staking.unbond(UNBOND_VALUE)
+      .signAndSend(alicePair, ({ events = [], status }) => {
+        expect(status.type).toMatch(/Ready/i);
+      });
+    });
+
+    it('verifies that derive.staking.unlocking isn\'t empty/undefined', () => {
+      return api.derive.session.info(ALICE_STASH, (info: DerivedStaking) => {
+        expect(info.unlocking).toBeDefined();
```

Comment:
Some tweaks, adjust as needed -

```js
expect(info.unlocking).toHaveLength(1);
expect(info.unlocking[0].value.eqn(UNBOND_VALUE)).toBe(true);
```

Reply:
The thing here is that `unlocking` could have a length bigger than `1` ( e.g if you've unbonded before, like you've run the test before, or if you have redeemable funds).
You can't be sure about any `unlocking[x].value.eqn` because the value is a sum (in case you've unbonded several times in the same session, you might have a value>`UNBOND_VALUE`). Also you can't be sure about what x is because things aren't ordered anymore. Just did a test with a change of bonding duration and although the extrinsic keeps the order of when the events came in, things get shuffled because of the derive calculation.

I'll use `expect(info.unlocking).toBeGreaterThan(0)` which should be the same as `.toBeDefined()` considering how we built the derive.

Reply:
That is fine, just comment it with "this needs a clean chain"